ranger/tmpl/layout/vue.tmpl

```diff
@@ -7,7 +7,11 @@
     var InitialVueProps = {{ .Data.props }}
   </script>
 
-  {{ packTag .Data.entry false }}
+  {{ if isDevelopment }}
+  <script src="{{ asset (print "src/pages/" .Data.entry ".ts") }}" type="module"></script>
+  {{ else }}
+  <script src="{{ asset (print "assets/" .Data.entry ".js") }}" type="module"></script>
+  {{ end }}
```

Comment:
I accept that this might be an unexpected change, but once I landed on it, I felt like there was improved clarity on the problem that is, the entry point exists at a different path when the development mode is changed, and it avoids needing to "configure" the sub directory via Trails or Vite.  They both just need to be configured to serve their respective assets under `client/dist`.

It also avoids hiding away special handling for file extensions. i.e. `.ts` vs `.js`, and while we don't do it, a similar case for could exist for `.scss` and `.css`.  

Finally, FWIW, once I started using a tool chain that came with a dev server (browserify, webpack dev server, vite serve), I've always ended up keeping this logic in my template.

Reply:
Waffling on whether this should get hidden in trails to keep the implementation simpler in the tmpl, but I think I agree that it puts the problem/solution in plain view which is probably the right call.

Reply:
Yeah, I tried it a few ways., but this one out in part because we aren't putting a lot of asset related tags into the templates anyway.  It also helps decouple from the client a bit more, giving a Trails user a little more freedom to structure their client src as preferred.  With a configurable origin, it goes further and opens up the flexibility to use Trails with a client tool chain other than Vite since the only thing Vite specific becomes the pattern match for hashed assets.